direct/common/_poisson.pyx

```diff
@@ -0,0 +1,110 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+#cython: overflowcheck=False
+#cython: unraisable_tracebacks=False
+
+import numpy as np
+cimport numpy as cnp
+from libc.math cimport cos, pi, sin
+from libc.stdlib cimport RAND_MAX, rand, srand
+
+cnp.import_array()
+
+
+cdef double random_uniform() nogil:
+    """Produces a random number in (0, 1)."""
+    cdef double r = rand()
+    return r / RAND_MAX
+
+
+cdef int randint(int upper) nogil:
+    """Produces a random integer in {0, 1, ..., upper-1}."""
+    return int(random_uniform() * (upper))
+
+
+cdef inline Py_ssize_t fmax(Py_ssize_t one, Py_ssize_t two) nogil:
+    """Max(a, b)."""
+    return one if one > two else two
+
+
+cdef inline Py_ssize_t fmin(Py_ssize_t one, Py_ssize_t two) nogil:
+    """Min(a, b)."""
+    return one if one < two else two
+
+
+def poisson(
+    int nx,
+    int ny,
+    int max_attempts,
+    cnp.ndarray[cnp.int_t, ndim=2, mode='c'] mask,
```

Comment:
This is based on some code right? You should reference this, and see if the licenses are compatible